src/app/services/share.js

```diff
@@ -132,6 +122,115 @@ module.exports = (Model, App) => {
     return newShare.token;
   };
 
+  /**
+   * Generates a share folder token
+   * @param user
+   * @param folderId
+   * @param bucket
+   * @param mnemonic
+   * @param fileToken
+   * @param views
+   * @returns {Promise<string|*>}
+   * @constructor
+   */
+  const GenerateFolderTokenAndCode = async (
+    user,
+    folderId,
+    bucket,
+    mnemonic,
+    fileToken,
+    views = 1,
+  ) => {
+    const itemExists = await Model.folder.findOne({
+      where: {
+        id: { [Op.eq]: folderId },
+        user_id: { [Op.eq]: user.id },
+      }
+    });
+
+    if (!itemExists) {
+      throw Error('Folder not found');
+    }
+
+    const maxAcceptableSize = 1024 * 1024 * 1000; // 1GB
+
+    const folderSize = await getFolderSize(folderId);
+
+    if (folderSize > maxAcceptableSize) {
+      throw Error('Folder too large');
+    }
+
+    // Generate password
+    const code = crypto.randomBytes(32).toString('hex');
+
+    // Encrypt mnemonic with password
+    const encryptedMnemonic = encrypt(mnemonic, code);
+
+    // Generate a new share token
+    const newToken = crypto.randomBytes(10).toString('hex');
+
+    const share = await Model.shares.findOne({
+      where: {
+        file: {
+          [Op.eq]: folderId
+        },
+        user: {
+          [Op.eq]: user.email
+        }
+      },
+    });
+
+    if (share) {
+      // Update share details
+      Model.shares.update(
+        {
+          token: newToken,
+          mnemonic: encryptedMnemonic,
+          isFolder: true,
+          views: views,
+          fileToken: fileToken,
+        },
+        {
+          where: {
+            id: {
+              [Op.eq]: share.id
+            }
+          }
+        },
+      );
+    } else {
+      // Create share details
+      await Model.shares.create({
+        token: newToken,
+        mnemonic: encryptedMnemonic,
+        encryptionKey: '',
+        file: folderId,
+        user: user.email,
+        isFolder: true,
+        views: views,
+        bucket: bucket,
+        fileToken: fileToken,
+      });
+    }
+
+    return {
+      token: newToken,
+      code: code
+    };
+  };
+
+  const encrypt = (text, key) => {
```

Comment:
Use `crypto`, avoid using `crypto-js` at all costs, we have it for retro-compatibility, but we rely on `crypto` Node.js module as `crypto-js` has given problems in the past due to a bug that makes some random encrypted texts only decryptable with `crypto-js`. Also, in @internxt/lib, you have [aes](https://github.com/internxt/lib/blob/master/src/aes/encrypt.ts) already implemented.